packages/iov-core/src/transactionencoder.ts

```diff
@@ -0,0 +1,101 @@
+import { Encoding } from "@iov/encoding";
+import { JsonCompatibleValue } from "@iov/jsonrpc";
+
+/**
+ * Encodes non-circular JavaScript objects and primitives into JSON.
+ * Used for encoding/decoding transactions but works for kind of data consisting of the supported types.
+ *
+ * Supported types:
+ * - boolean
+ * - number
+ * - null
+ * - object
+ * - Array
+ * - string
+ * - Uint8Array
+ */
+export class TransactionEncoder {
+  public static toJson(data: unknown): JsonCompatibleValue {
+    if (typeof data === "number" || typeof data === "boolean") {
+      return data;
+    }
+
+    if (data === null) {
+      return null;
+    }
+
+    if (typeof data === "string") {
+      return `string:${data}`;
+    }
+
+    if (data instanceof Uint8Array) {
+      return `bytes:${Encoding.toHex(data)}`;
+    }
+
+    if (Array.isArray(data)) {
+      return data.map(TransactionEncoder.toJson);
+    }
+
+    // Exclude special kind of objects like Array, Date or Uint8Array
+    // Object.prototype.toString() returns a specified value:
+    // http://www.ecma-international.org/ecma-262/7.0/index.html#sec-object.prototype.tostring
+    if (
+      typeof data === "object" &&
+      data !== null &&
+      Object.prototype.toString.call(data) === "[object Object]"
+    ) {
+      const out: any = {};
+      for (const key of Object.keys(data)) {
+        // tslint:disable-next-line: no-object-mutation
+        out[key] = TransactionEncoder.toJson((data as any)[key]);
```

Comment:
`data as Object`?

Reply:
Gives this error

> Element implicitly has an 'any' type because type 'Object' has no index signature.

Reply:
In my experiance, the `object` type is pretty useless as it does not mean _any kind of object I don't know more about_, but basically means `{}`. I don't know the difference between `object` and `Object` but I read once that there are only very few cases where `Object` or `String` should be used as types.